deepspeed/module_inject/replace_module.py

```diff
@@ -464,20 +464,28 @@ def _slice_embedding(child, name, conv_linear_layer):
 
         def update_mp_params(child):
             if hasattr(child, 'n_heads'):
+                assert child.n_heads%mp_size == 0, "Model params must be divisible by mp_size"
```

Comment:
The following changes would make the assert message more meaningful and actionable for users:
1. Include the actual dimension (e.g., hidden size) that violates the divisibility requirement. 
2. Include the invalid dimension value.
3. Report all the failing dimensions in one message (optional).